lib/SILOptimizer/Transforms/SimplifyCFG.cpp

```diff
@@ -2096,29 +2090,27 @@ bool SimplifyCFG::simplifySwitchEnumBlock(SwitchEnumInst *SEI) {
   LLVM_DEBUG(llvm::dbgs() << "fold switch " << *SEI);
 
   auto *EI = dyn_cast<EnumInst>(SEI->getOperand());
+  auto loc = SEI->getLoc();
   SILBuilderWithScope Builder(SEI);
   if (!LiveBlock->args_empty()) {
     SILValue PayLoad;
     if (SEI->hasDefault() && LiveBlock == SEI->getDefaultBB()) {
       assert(Fn.hasOwnership() && "Only OSSA default case has an argument");
       PayLoad = SEI->getOperand();
     } else {
-      if (EI) {
-        PayLoad = EI->getOperand();
```

Comment:
I got rid of it here since this exists in SILCombine anyway. Only because, this needed some complex condition checking. Eg.:  In OSSA, enum instruction can be ForwardingConsume and we can directly use the enum's operand only if it going to get deleted later on.